libp2p/transports/tls/certificate.nim

```diff
@@ -0,0 +1,489 @@
+# Nim-LibP2P
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import std/[sequtils, strutils]
+
+import stew/byteutils
+import chronicles
+
+import mbedtls/pk
+import mbedtls/ctr_drbg
+import mbedtls/entropy
+import mbedtls/ecp
+import mbedtls/sha256
+import mbedtls/md
+import mbedtls/asn1
+import mbedtls/asn1write
+import mbedtls/x509
+import mbedtls/x509_crt
+import mbedtls/oid
+import mbedtls/debug
+import mbedtls/error
+import nimcrypto/utils
+import ../../crypto/crypto
+
+logScope:
+  topics = "libp2p tls certificate"
+
+# Constants and OIDs
+const
+  P2P_SIGNING_PREFIX = "libp2p-tls-handshake:"
+  SIGNATURE_ALG = MBEDTLS_MD_SHA256
+  EC_GROUP_ID = MBEDTLS_ECP_DP_SECP256R1
+  LIBP2P_EXT_OID_DER: array[10, byte] =
+    [0x2B, 0x06, 0x01, 0x04, 0x01, 0x83, 0xA2, 0x5A, 0x01, 0x01]
+    # "1.3.6.1.4.1.53594.1.1"
+
+# Exception types for TLS certificate errors
+type
+  TLSCertificateError* = object of Exception
```

Comment:
As it is inside nim-libp2p, maybe inherit from LPError instead of exception?

Reply:
done.